pkg/factory/risingwave_object_factory.go

```diff
@@ -1575,12 +1583,47 @@ func (f *RisingWaveObjectFactory) portsForFrontendContainer() []corev1.Container
 	}
 }
 
+func (f *RisingWaveObjectFactory) isEmbeddedServingModeEnabled() bool {
+	return ptr.Deref(f.risingwave.Spec.EnableEmbeddedServingMode, false)
+}
+
 func (f *RisingWaveObjectFactory) setupFrontendContainer(container *corev1.Container) {
 	container.Name = "frontend"
-	container.Args = []string{"frontend-node"}
+
+	if f.isEmbeddedServingModeEnabled() {
+		container.Args = []string{
+			"standalone",
```

Comment:
I'm curious when running standalone mode, will it reserve unnecessary resources for the standalone meta service/compactor service causing waste?

Reply:
`standalone` will start services depending on the arguments. In this case, only `--frontend-opts` and `compute-opts` are passed thus meta and compactor won't be started at all. However, there's a concrete problem from the competition of memory between the frontend and serving compute. IMO, it should not be a problem for now since the compute node (whether in serving or streaming mode) will reserve quite a lot memory for system. And the reserved memory should be quite enough for frontend to run. But it's still a problem needs to be addressed as  @chenzl25 said in the Slack [discussion](https://risingwave-labs.slack.com/archives/C036F5Z3EMD/p1715594516286529?thread_ts=1715591650.996399&cid=C036F5Z3EMD).